### PR TITLE
[1LP][RFR]New Test: test_reports_menu_with_duplicate_reports

### PR DESCRIPTION
**Pull request by valaparthvi**
## Purpose or Intent

- __Extending__ 
    1. `get_custom_report` fixture to import a report instead of creating a new report with the yaml data. 
    2. `rbac_api` to not delete user, since that is anyway done by `gen_data._create_skeleton`.
- __Adding tests__ `test_reports_menu_with_duplicate_reports` which automates BZ 1676638, 1731017
- __Updating tests__ `test_rbac_move_custom_report` and `test_add_reports_to_available_reports_menu` to use the updated `get_custom_report` fixture.

### PRT Run
{{ pytest: cfme/tests/intelligence/reports/test_menus.py -k "test_add_reports_to_available_reports_menu or test_rbac_move_custom_report or test_reports_menu_with_duplicate_reports" -vvv }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit ec500c31885e2d0005de323fa232720baae7dae6

The *local* fixture **`get_custom_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

The *local* fixture **`rbac_user`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/intelligence/reports/test_menus.py [hunk 1]
```diff
@@ -171,3 +169,58 @@ def test_rbac_move_custom_report(
...
+    report_menus.move_reports(group, folder, subfolder, custom_report_1.menu_name)
+
+    # instantiate the expected report and check for its existence
+    expected_report = appliance.collections.reports.instantiate(
+        type=folder, subtype=subfolder, menu_name=custom_report_1.menu_name
+    )
+    assert expected_report.exists
+
+    # delete the first report and check if the expected report exists
+    custom_report_1.delete()
```
Comment (sbulage) [inline cfme/tests/intelligence/reports/test_menus.py]: Suggestion:
You can use `delete()` operation inside (`request.addfinalizer(custom_report_2.delete)`)

File: cfme/tests/intelligence/reports/test_menus.py [hunk 2]
```diff
@@ -171,3 +169,58 @@ def test_rbac_move_custom_report(
...
+        expectedResults:
+            1. Report must be visible.
+            2.
+            3. Report must not be visible.
+            4.
+            5. Report must not be visible.
+    Bugzilla:
+        1731017
+        1676638
+    """
```
Comment (ganeshhubale) [inline cfme/tests/intelligence/reports/test_menus.py]: @valaparthvi 
**Optional:**
- It will be better if you put ```Bugzilla``` block above ```Polarion``` data.
- In my observation, Bugzilla numbers are not getting updated in polarion you keep it as it is.
Please check your old polarion test cases once and decide.

File: cfme/tests/intelligence/reports/test_menus.py [hunk 3]
```diff
@@ -171,3 +169,58 @@ def test_rbac_move_custom_report(
...
+def test_reports_menu_with_duplicate_reports(appliance, request, group, report_menus):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        initialEstimate: 1/10h
+        startsin: 5.9
+        setup:
+            1.Create a custom report or copy an existing report.
+            2. Move the custom report to 'Tenants > Tenant Quotas' menu in user's current group
```
Comment (ganeshhubale) [inline cfme/tests/intelligence/reports/test_menus.py]: - Need space here:  ```1. Create a custom .....```
- Also I can see this test case ignored for ```5.10``` but it's ```startsin``` version is ```5.9```. I think it should be ```5.11```

File: cfme/tests/intelligence/reports/test_menus.py [hunk 4]
```diff
@@ -132,17 +133,14 @@ def test_add_reports_to_available_reports_menu(
 @pytest.fixture()
 def rbac_user(appliance, request, group):
     user, user_data = _users(request, appliance, group=group)
-    yield appliance.collections.users.instantiate(
+    return appliance.collections.users.instantiate(
         name=user[0].name,
```
Comment (ganeshhubale) [inline cfme/tests/intelligence/reports/test_menus.py]: Question:
Why did you change it to ```return```?

Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_menus.py]: Because I am not going to come back to this fixture for any cleanup, so I just return the values.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 09f324daf3e20c94a8c143b99a3ce8394f7104f9

The *local* fixture **`get_custom_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

The *local* fixture **`rbac_user`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/intelligence/reports/test_menus.py (hunk 1 above)
Comment (ganeshhubale) [reply cfme/tests/intelligence/reports/test_menus.py]: - I think this is the part of test case. If test fails to delete the ```custom_report_1``` then this test should fail.
- So according to me, we don't need ```request.addfinalizer(custom_report_1.delete)```

What do you think @valaparthvi ?

Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_menus.py]: Hmm, that does seem like a valid point :sweat_smile:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit ed81a04f0dec003b1025e1ddc4fbf3994aed097a

The *local* fixture **`get_custom_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

The *local* fixture **`rbac_user`** is used in the following files:
- **cfme/tests/intelligence/reports/test_menus.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/intelligence/reports/test_menus.py (hunk 1 above)
Comment (sbulage) [reply cfme/tests/intelligence/reports/test_menus.py]: I think then finalizer not is require for copied(`custom_report_2`) report, simply `delete()` will do the needful.?
@valaparthvi can you please confirm this .? Just Curious :smile:

Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_menus.py]: Deleting `custom_report_1` was a part of the test, but deleting `custom_report_2` is not, so it's better to delete `custom_report_2` as a part of the cleanup.

File: cfme/tests/intelligence/reports/test_menus.py [hunk 5]
```diff
@@ -171,3 +169,60 @@ def test_rbac_move_custom_report(
...
+    ).copy()
+
+    folder, subfolder = "Tenants", "Tenant Quotas"
+    report_menus.move_reports(group, folder, subfolder, custom_report_1.menu_name)
+
+    # instantiate the expected report and check for its existence
+    expected_report = appliance.collections.reports.instantiate(
+        type=folder, subtype=subfolder, menu_name=custom_report_1.menu_name
+    )
+    assert expected_report.exists
```
Comment (sbulage) [inline cfme/tests/intelligence/reports/test_menus.py]: Good PR :100: 
Like this implementation. :+1:
